### PR TITLE
Test: Delete events on identity

### PR DESCRIPTION
**Pull request by eloycoto**
Two test to validate the delete events on identities, read per commit.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7472)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/chaos.go [hunk 1]
```diff
@@ -194,4 +195,34 @@ var _ = Describe("RuntimeChaos", func() {
 			check.res.ExpectSuccess("Curl from app2 to httpd1 should work but it failed at %s", check.time)
 		}
 	})
+
+	It("Validate that delete events on KVStore do not release in use identities", func() {
+		// This validates that if a kvstore delete event is send the identity
+		// is not release if it is in use. For more info issue #7240
+
+		prefix := "http://127.0.0.1:8500/v1/kv/cilium/state/identities/v1/id"
```
Comment (ianvernon) [inline test/runtime/chaos.go:203]: Is there a way we can get this prefix in the kvstore from the package itself? Not sure if that'd require factoring certain stuff into a 'constants' package within the kvstore, though.

File: test/runtime/chaos.go [hunk 2]
```diff
@@ -194,4 +195,34 @@ var _ = Describe("RuntimeChaos", func() {
...
+			"Identities are not the same after delete keys from kvstore")
+
+		for _, identityID := range newidentities {
+			id, err := identity.ParseNumericIdentity(identityID)
+			Expect(err).To(BeNil(), "Cannot parse identity")
+			if id.IsReservedIdentity() {
+				continue
+			}
+			action := helpers.CurlFail("%s/%s", prefix, identityID)
+			vm.Exec(action).ExpectSuccess("Key %s cannot is not restored correctly", identityID)
```
Comment (ianvernon) [inline test/runtime/chaos.go:225]: I'd change this to say "Key %s was deleted when it should not have been"

**Review by ianvernon — APPROVED** [review]
Nothing blocking. Good idea to add a test for this!

**Pull request merged**